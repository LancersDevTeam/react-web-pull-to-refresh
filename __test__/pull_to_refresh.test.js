import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import PullToRefresh from '../src/pull_to_refresh';
import renderer from 'react-test-renderer';

describe('PullToRefresh', () => {
    test('renders correctly', () => {
        const pullToRefreshProps = {
            pullUpToRefresh: {
                displayFlag: true,
                pullUpHandler: () => {},
                loadingContent: 'Loading...',
            },
            pullDownToRefresh: {
                displayFlag: false,
                pullDownHandler: () => {},
                loadingContent: 'Loading...',
            }
        }
        const list = [];
        for (let i = 0; i < 10; i++) {
            list.push(<li key={i}>{i}</li>);
        }
        const tree = mountToJson(mount(
            <PullToRefresh {...pullToRefreshProps}>
                <ul>{list}</ul>
            </PullToRefresh>
        ));
        expect(tree).toMatchSnapshot();
    });
});
