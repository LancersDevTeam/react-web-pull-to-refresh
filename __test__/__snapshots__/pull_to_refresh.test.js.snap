// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullToRefresh renders correctly 1`] = `
<PullToRefresh
  pullDownToRefresh={
    Object {
      "displayFlag": false,
      "loadingContent": "Loading...",
      "pullDownHandler": [Function],
    }
  }
  pullUpToRefresh={
    Object {
      "displayFlag": true,
      "loadingContent": "Loading...",
      "pullUpHandler": [Function],
    }
  }
>
  <PullUpToRefresh
    backgroundColor="#f2f2f2"
    borderHeight={0}
    className=""
    color="black"
    displayFlag={true}
    fontSize="20px"
    height={50}
    loadingContent="Loading..."
    margin="0px"
    padding="0px"
    pullUpHandler={[Function]}
    threshold={10}
    throttleTermMS={80}
  >
    <div
      className=""
    >
      <div>
        <PullDownToRefresh
          backgroundColor="#f2f2f2"
          borderHeight={0}
          className=""
          color="black"
          displayFlag={false}
          fontSize="20px"
          height={50}
          loadingContent="Loading..."
          margin="0px"
          padding="0px"
          pullDownHandler={[Function]}
          threshold={10}
          throttleTermMS={80}
        >
          <div
            className=""
          >
            <LoadingArea
              backgroundColor="#f2f2f2"
              borderHeight={0}
              className=""
              color="black"
              displayFlag={false}
              fontSize="20px"
              height={50}
              loadingContent="Loading..."
              margin="0px"
              padding="0px"
              pullDownHandler={[Function]}
              threshold={10}
              throttleTermMS={80}
              updatingFlag={false}
            >
              <div
                className="loading_base not_loading"
              />
            </LoadingArea>
            <div>
              <ul>
                <li>
                  0
                </li>
                <li>
                  1
                </li>
                <li>
                  2
                </li>
                <li>
                  3
                </li>
                <li>
                  4
                </li>
                <li>
                  5
                </li>
                <li>
                  6
                </li>
                <li>
                  7
                </li>
                <li>
                  8
                </li>
                <li>
                  9
                </li>
              </ul>
            </div>
          </div>
        </PullDownToRefresh>
      </div>
      <LoadingArea
        backgroundColor="#f2f2f2"
        borderHeight={0}
        className=""
        color="black"
        displayFlag={true}
        fontSize="20px"
        height={50}
        loadingContent="Loading..."
        margin="0px"
        padding="0px"
        pullUpHandler={[Function]}
        threshold={10}
        throttleTermMS={80}
        updatingFlag={false}
      >
        <div
          className="loading_base not_loading"
        >
          Loading...
        </div>
      </LoadingArea>
    </div>
  </PullUpToRefresh>
</PullToRefresh>
`;
